torch/csrc/jit/passes/fuse_linear.cpp

```diff
@@ -21,10 +21,22 @@ void FuseLinear(std::shared_ptr<Graph>& graph) {
     return is_int_constant(match, vmap, "beta", 1);
   };
 
+  // check %weight_t is produced by `aten::t` to make sure
```

Comment:
This feels brittle, what if the user already has a weight that is transposed already?

Reply:
we don't transform it to `aten::linear` in that case since we don't know the dimension of weight. we can implement quantized::matmul if we need to quantize it